projects/fsastorefrontlib/src/core/my-account/services/policy-chart-data.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { EChartsOption } from 'echarts';
+
+@Injectable()
+export class PolicyChartDataService {
+  protected readonly SEPARATOR = '.';
+
+  groupPoliciesByAttribute(policies: any[], key: string | string[]) {
+    return policies.reduce((policy, item) => {
+      const groupCriteria = this.getObjectValueByProperty(key, item);
+      policy[groupCriteria] = policy[groupCriteria] || [];
+      policy[groupCriteria].push({
+        categoryData: item.categoryData,
+        policyPremium: item.policyPremium,
+        paymentFrequency: item.paymentFrequency,
+      });
+      return policy;
+    }, {});
+  }
+
+  calculatePremiumAmountByCategory(policies, chartOption: EChartsOption) {
+    chartOption.series[0].data = [];
+    Object.keys(policies).forEach(category => {
+      const reducer = (sum, policy) => sum + policy.policyPremium.value;
+      const initialValue = 0;
+      const policyAmountByCategory = policies[category].reduce(
+        reducer,
+        initialValue
+      );
+      if (policyAmountByCategory) {
+        chartOption.series[0].data.push({
+          value: policyAmountByCategory,
+          name: category.split(' ')[0],
+          currancyValue: policyAmountByCategory.toLocaleString('de-DE', {
```

Comment:
1. currancyValue -> currencyValue. 
2. Fixing 'de-DE' doesn't seem like a good idea. Perhaps we can pass language dynamically, maybe via `languageService.getActive()` or something like that

Reply:
if I use `languageService.getActive()` than different currency format depends on the language, e.g. for `en` language will be `€ 150` and for `de` will be `150 €`. If it is suitable behaviour I can apply that.